### PR TITLE
(feat) O3-3183: ID number labels should not be tags

### PR DESCRIPTION
**Pull request by Twiineenock**
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [ ] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [ ] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

Only the name of the primary identifier for a patient should be held in a tag like this. All other ID numbers display as label
A deep description here [ID number labels should not be tags](https://openmrs.atlassian.net/browse/O3-3183)

The configured code for a primary Identifier is the value of the metadataUuid from this end-point `${restBaseUrl}/metadatamapping/termmapping?v=custom:(metadataUuid)&code=emr.primaryIdentifierType` 
which return this `{"results":[{"metadataUuid":"05a29f94-c0ed-11e2-94be-8c13b969e334"}]}`

With this PR all identifiers whose code is equal to the metadataUuid, are regarded as primary identifiers and their labels held in tags.
### Example of patient identifiers
[code.json](https://github.com/user-attachments/files/15573892/code.json)

Note: the only primary Identifier here is the OpenMRS ID because its code: ` "code": "05a29f94-c0ed-11e2-94be-8c13b969e334"` is the metadataUuid: `"metadataUuid":"05a29f94-c0ed-11e2-94be-8c13b969e334"` The Old Identification Number and ID Card are not.

[Patient header design docs](https://zeroheight.com/23a080e38/p/6663f3-patient-header)

### After fix
banner

## Related Issue

[https://openmrs.atlassian.net/browse/O3-3183](https://openmrs.atlassian.net/browse/O3-3183)

### CODE REVIEW

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx [hunk 1]
```diff
@@ -41,12 +44,20 @@ export function PatientBannerPatientInfo({ patient }: PatientBannerPatientInfoPr
...
           {filteredIdentifiers?.length
-            ? filteredIdentifiers.map(({ value, type }) => (
+            ? filteredIdentifiers.map(({ value, type }, index) => (
                 <span key={value} className={styles.identifierTag}>
-                  <Tag className={styles.tag} type="gray" title={type?.text}>
-                    {type?.text}
-                  </Tag>
-                  {value}
+                  <div>{index > 0 && <span className={styles.separator}>&middot;</span>}</div>
+                  {type?.coding?.[0]?.code == primaryIdentifierCode ? (
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx:50]:
```suggestion
                  {type?.coding?.[0]?.code === primaryIdentifierCode ? (
```

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx [hunk 2]
```diff
@@ -41,12 +44,20 @@ export function PatientBannerPatientInfo({ patient }: PatientBannerPatientInfoPr
...
+                    <div className={styles.tag}>
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx:51]:
```suggestion
                    <div className={styles.primaryIdentifier}>
```

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx [hunk 3]
```diff
@@ -41,12 +44,20 @@ export function PatientBannerPatientInfo({ patient }: PatientBannerPatientInfoPr
...
+                      <Tag type="gray" title={type?.text}>{`${type?.text}:`}</Tag>
+                      <span>{`${value}`}</span>
+                    </div>
+                  ) : (
+                    <label htmlFor="identifier" className={styles.noTag}>
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx:56]:
```suggestion
                    <label htmlFor="identifier" className={styles.secondaryIdentifier}>
```

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss [hunk 4]
```diff
@@ -100,11 +100,31 @@
...
 }
 
 .tag {
-  margin: 0.25rem 0.25rem 0.25rem 0rem;
+  display: flex;
+  align-items: center;
+  gap: 0.25rem;
+}
+
+.noTag {
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss:116]:
```suggestion
.secondaryIdentifier {
```

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/primary-identifier.resource.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import useSWR from 'swr';
+import { type FetchResponse, restBaseUrl } from '@openmrs/esm-framework';
+import { openmrsFetch } from '@openmrs/esm-framework';
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/patient-banner/patient-info/primary-identifier.resource.tsx:3]: Consolidate these into one import.

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss [hunk 6]
```diff
@@ -100,11 +100,31 @@
 .identifierTag {
   display: flex;
   align-items: center;
+}
+
+.separator {
+  margin-left: 0.75rem;
   margin-right: 0.75rem;
 }
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss:109]: The `.identifiers` class should have its color property set to:

```color: colors.$gray-70;```

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss [hunk 7]
```diff
@@ -100,11 +100,31 @@
...
+  display: flex;
+  align-items: center;
+  gap: 0.25rem;
+}
+
+.noTag {
+  display: flex;
+
+  .label {
+    font-size: 12px;
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss:120]:
```suggestion
     @include type.type-style('label-01');
```

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss [hunk 8]
```diff
@@ -100,11 +100,31 @@
...
+    margin: 0.25rem 0.25rem 0.25rem 0rem;
+    align-self: auto;
+  }
+
+  .identifier {
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss:125]:
```suggestion
  .identifier {
      @include type.type-style('body-compact-02');
```

**Review by denniskigen — CHANGES_REQUESTED** [review]
Thanks, @Twiineenock. It is a good start, but a few things need to be aligned better with the designs.

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/primary-identifier.resource.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+interface PrimaryIdentifierResponse {
+  results: Array<PrimaryIdentifier>;
+}
+
+export function usePrimaryIdentifierCode(): {
+  primaryIdentifierCode: string | undefined;
+  isLoading: boolean;
+} {
+  const url = `${restBaseUrl}/metadatamapping/termmapping?v=custom:(metadataUuid)&code=emr.primaryIdentifierType`;
+  const { data, error } = useSWR<FetchResponse<PrimaryIdentifierResponse>, Error>(url, openmrsFetch);
```
[the remark this answers is not in the document]
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/primary-identifier.resource.tsx]: Thanks @denniskigen for your time invested in giving a comprehensive review. I have updated this PR to meet all the requested changes except this one. 
useSWR
 
I get this error when I hover over `(primaryIdentifierCode ? url : null, openmrsFetch)`:
`Cannot find name 'primaryIdentifierCode'. Did you mean 'usePrimaryIdentifierCode'?`.
And is useSWR able to know that respose body does not contain a given value before making the request?
Thanks!

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/primary-identifier.resource.tsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  const memoisedPrimaryIdentifier = useMemo(
+    () => ({
+      primaryIdentifierCode: data?.data.results[0].metadataUuid,
+      isLoading: !data && !error,
+    }),
+    [data, error],
+  );
```
[the remark this answers is not in the document]
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/primary-identifier.resource.tsx]: Done thanks!

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/primary-identifier.resource.tsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+export interface PrimaryIdentifier {
+  metadataUuid: string;
+}
+interface PrimaryIdentifierResponse {
+  results: Array<PrimaryIdentifier>;
+}
+
+export function usePrimaryIdentifierCode(): {
+  primaryIdentifierCode: string | undefined;
+  isLoading: boolean;
```
[the remark this answers is not in the document]
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/primary-identifier.resource.tsx]: Done thanks!

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx [hunk 12]
```diff
@@ -41,12 +44,20 @@ export function PatientBannerPatientInfo({ patient }: PatientBannerPatientInfoPr
...
                 <span key={value} className={styles.identifierTag}>
-                  <Tag className={styles.tag} type="gray" title={type?.text}>
-                    {type?.text}
-                  </Tag>
-                  {value}
+                  <div>{index > 0 && <span className={styles.separator}>&middot;</span>}</div>
+                  {type?.coding?.[0]?.code == primaryIdentifierCode ? (
+                    <div className={styles.tag}>
+                      <Tag type="gray" title={type?.text}>{`${type?.text}:`}</Tag>
+                      <span>{`${value}`}</span>
```
[the remark this answers is not in the document]
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx]: Done thanks!

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss [hunk 13]
```diff
@@ -100,11 +100,31 @@
 .identifierTag {
   display: flex;
   align-items: center;
+}
+
+.separator {
+  margin-left: 0.75rem;
   margin-right: 0.75rem;
```
[the remark this answers is not in the document]
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss]: Done thanks!

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx (hunk 1 above)
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx]: Done thanks!

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx (hunk 2 above)
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx]: Done thanks!

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx (hunk 3 above)
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.component.tsx]: Done thanks!

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss (hunk 4 above)
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss]: Done thanks!

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss (hunk 7 above)
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss:119]: Done thanks!

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss (hunk 8 above)
Comment (Twiineenock) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/patient-banner-patient-info.module.scss:125]: Done thanks @denniskigen !

File: packages/framework/esm-styleguide/src/patient-banner/patient-info/primary-identifier.resource.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (denniskigen) [reply packages/framework/esm-styleguide/src/patient-banner/patient-info/primary-identifier.resource.tsx]: Yeah, you can safely ignore this.

**Comment by denniskigen** [comment]
@ibacher, is it obvious to you what the test failure is about?

**Comment by ibacher** [comment]
Not off-hand. I would move the data-fetching hook into react-utils though and maybe try not importing from `@openmrs/esm-framework`...

I suspect the Jest configuration is broken for that case.

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks, @Twiineenock. Thanks, @ibacher, for the fix suggestion. Worked a treat.

**Pull request merged**